adminapi/api.py

```diff
@@ -37,28 +22,10 @@ def _api_function(*args, **kwargs):
 
             result = send_request(API_CALL_ENDPOINT, call)
 
-            if result['status'] == 'success':
-                return result['retval']
-
             if result['status'] == 'error':
+                raise ApiError(result['message'])
```

Comment:
Maybe you can also provide information about the type of error in this `ApiError`. Passing `result['type']` to the `ApiError` would be worth considering.